gui/src/3D/controls/slider3D.ts

```diff
@@ -0,0 +1,246 @@
+import { Nullable } from "babylonjs/types";
+import { Observable } from "babylonjs/Misc/observable";
+import { Vector3 } from "babylonjs/Maths/math.vector";
+import { TransformNode } from "babylonjs/Meshes/transformNode";
+import { Scene } from "babylonjs/scene";
+import { Mesh } from "babylonjs/Meshes/mesh";
+import { Control3D } from "./control3D";
+import { MeshBuilder } from "babylonjs/Meshes/meshBuilder";
+import { PointerDragBehavior } from "babylonjs/Behaviors/Meshes/pointerDragBehavior";
+import { StandardMaterial } from "babylonjs/Materials/standardMaterial";
+import { Color3 } from "babylonjs/Maths/math.color";
+import { AbstractMesh } from "babylonjs/Meshes/abstractMesh";
+
+const SLIDER_MIN: number = 0;
+const SLIDER_MAX: number = 100;
+const SLIDER_VAL: number = 50;
+const SLIDER_STEP: number = 0;
+const SLIDER_SCALING: number = 2.0;
+
+export interface ISlider3DOptions {
+    minimum?: number;
+    maximum?: number;
+    value?: number;
+    step?: number;
+}
+
+/**
+ * Class used to create a slider in 3D
+ */
+export class Slider3D extends Control3D {
+    private _sliderBarMaterial: StandardMaterial;
+    private _sliderThumbMaterial: StandardMaterial;
+    private _sliderThumb: Mesh;
+    private _sliderBar: Mesh;
+
+    private _minimum: number;
+    private _maximum: number;
+    private _value: number;
+    private _step: number;
+
+    /** Observable raised when the sldier value changes */
+    public onValueChangedObservable = new Observable<number>();
+
+    /**
+     * Creates a new slider
+     * @param name defines the control name
+     */
+    constructor(name?: string, options?: ISlider3DOptions) {
+        super(name);
+
+        this.minimum = options?.minimum ? options.minimum : SLIDER_MIN;
+        this.maximum = options?.maximum ? options.maximum : SLIDER_MAX;
+        this.step = options?.step ? options.step : SLIDER_STEP;
+        this.value = options?.value ? options.value : SLIDER_VAL;
+    }
+
+    /**
+     * Gets the mesh used to render this control
+     */
+    public get mesh(): Nullable<AbstractMesh> {
+        if (this.node) {
+            return this._sliderThumb;
+        }
+
+        return null;
+    }
+
+    /** Gets or sets minimum value */
+    public get minimum(): number {
+        return this._minimum;
+    }
+
+    public set minimum(value: number) {
+        if (this._minimum === value) {
+            return;
+        }
+
+        this._minimum = Math.max(value, SLIDER_MIN);

```

Comment:
Ah, so the SLIDER_* values are used as limiting values, not just the initial state. Interesting. I want to say there should also be a check in here that the new minimum is not larger than the current maximum, but I remember talking already about how that check doesn't exist for the 2D slider...

Reply:
We are assuming the slider values are only positive numbers. Since the default min value is 0, we are using it here to limit the min value the user trying to set to the default min value. Since 2D slider don't prevent the user from setting min > max, I am assuming we can leave it like this. The slider won't crash but will not work as intended. I was trying to avoid this circular validation for min & max. It is very confusing for the end user.